packages/lib/src/helpers/kv_public_input.ts

```diff
@@ -0,0 +1,173 @@
+var EC = require("elliptic").ec;
+const BN = require("bn.js");
+
+import { bytesToBigInt, bigIntToBytes } from "./utils";
+import { EffECDSAPubInput } from "../types";
+
+const ec = new EC("secp256k1");
+
+const SECP256K1_N = new BN(
+  "fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141",
+  16
+);
+
+export class KVCircuitPubInput {
+  merkleRoot: bigint;
+  Tx: bigint;
+  Ty: bigint;
+  Ux: bigint;
+  Uy: bigint;
+  claimValue: bigint;
+  hashPubKeySecret: bigint;
+  hashSecret: bigint;
+
+  constructor(
+    merkleRoot: bigint,
+    Tx: bigint,
+    Ty: bigint,
+    Ux: bigint,
+    Uy: bigint,
+    claimValue: bigint,
+    hashPubKeySecret: bigint,
+    hashSecret: bigint
+  ) {
+    this.merkleRoot = merkleRoot;
+    this.Tx = Tx;
+    this.Ty = Ty;
+    this.Ux = Ux;
+    this.Uy = Uy;
+    this.claimValue = claimValue;
+    this.hashPubKeySecret = hashPubKeySecret;
+    this.hashSecret = hashSecret;
+  }
+
+  serialize(): Uint8Array {
+    let serialized = new Uint8Array(32 * 8)
+
+    serialized.set(bigIntToBytes(this.merkleRoot, 32), 0);
+    serialized.set(bigIntToBytes(this.Tx, 32), 32);
+    serialized.set(bigIntToBytes(this.Ty, 32), 64);
+    serialized.set(bigIntToBytes(this.Ux, 32), 96);
+    serialized.set(bigIntToBytes(this.Uy, 32), 128);
+    serialized.set(bigIntToBytes(this.claimValue, 32), 160);
+    serialized.set(bigIntToBytes(this.hashPubKeySecret, 32), 192);
+    serialized.set(bigIntToBytes(this.hashSecret, 32), 224);
+
+    return serialized;
+  }
+
+  static deserialize(serialized: Uint8Array): KVCircuitPubInput {
+    const merkleRoot = bytesToBigInt(serialized.slice(0, 32));
+    const Tx = bytesToBigInt(serialized.slice(32, 64));
+    const Ty = bytesToBigInt(serialized.slice(64, 96));
+    const Ux = bytesToBigInt(serialized.slice(96, 128));
+    const Uy = bytesToBigInt(serialized.slice(128, 160));
+    const claimValue = bytesToBigInt(serialized.slice(160, 192));
+    const hashPubKeySecret = bytesToBigInt(serialized.slice(192, 224));
+    const hashSecret = bytesToBigInt(serialized.slice(224, 256));
+
+    return new KVCircuitPubInput(merkleRoot, Tx, Ty, Ux, Uy, claimValue, hashPubKeySecret, hashSecret);
+  }
+}
+
+/**
+ * Public values of the membership circuit
+ */
+export class PublicInput {
+  r: bigint;
+  rV: bigint;
+  msgHash: Buffer;
+  circuitPubInput: KVCircuitPubInput;
+
+  constructor(
+    r: bigint,
+    v: bigint,
+    msgHash: Buffer,
+    circuitPubInput: KVCircuitPubInput
+  ) {
+    this.r = r;
+    this.rV = v;
+    this.msgHash = msgHash;
+    this.circuitPubInput = circuitPubInput;
+  }
+
+  serialize(): Uint8Array {
+    const circuitPubInput: Uint8Array = this.circuitPubInput.serialize();
+    let serialized = new Uint8Array(
+      32 + 1 + this.msgHash.byteLength + circuitPubInput.byteLength
+    );
+
+    serialized.set(bigIntToBytes(this.r, 32), 0);
+    serialized.set(bigIntToBytes(this.rV, 1), 32);
```

Comment:
could potentially make vars like:
```
const rLength = 32;
const rVLength = 1;

let serialized = new Uint8Array(
      rLength + rVLength + this.msgHash.byteLength + circuitPubInput.byteLength
    );

    serialized.set(bigIntToBytes(this.r, rLength), 0);
    serialized.set(bigIntToBytes(this.rV, rVLength), rLength);
...
```

Just to make it clear what each padding is for (but this is extremely minor)